packages/paima-sdk/paima-mw-core/src/wallets/evm/injected.ts

```diff
@@ -7,26 +7,40 @@ import { connectInjected } from '../wallet-modes.js';
 import { WalletMode } from '@paima/providers';
 import type { ApiForMode, IProvider } from '@paima/providers';
 import { EvmInjectedConnector } from '@paima/providers';
-import { GlobalConfig } from '@paima/utils';
+import { ConfigNetworkType, GlobalConfig } from '@paima/utils';
 
 interface SwitchError {
   code: number;
 }
 
-async function switchChain(): Promise<boolean> {
-  const mainEvmConfig = await GlobalConfig.mainEvmConfig();
+async function switchChain(chainName: string | undefined): Promise<boolean> {
```

Comment:
I think this may also close #336 but not sure.

Reply:
I think it does. We'll have to update the docs to mention this new feature